### PR TITLE
feat: add custom error message on signups disabled page

### LINKED ISSUE #11679: Customizable error page, when OIDC signup is disabled.
Would like a customizable error page, when OIDC signup is disabled.

> [!NOTE]
> See product requirements in the comment below

### PR DESCRIPTION
**Pull request by mtojek**
Fixes: https://github.com/coder/coder/issues/11679

Once Coder administrators sets this custom error message: 

```bash
export CODER_OIDC_SIGNUPS_DISABLED_TEXT='**Coder** is in early access mode. To request access, fill out this [form](https://internal.form.com).'
```

It will be rendered as follows:
Screenshot 2024-02-01 at 13 00 45

### CODE REVIEW

**Comment by matifali** [comment]
Thanks @mtojek. I think having a minimal markdown that can render links or make them clickable should be done.

**Comment by mtojek** [comment]
Alright, I'm going to expand it!

**Comment by mtojek** [comment]
@matifali Ok, please let me know your thoughts!

File: coderd/parameter/renderer_test.go [hunk 1]
```diff
@@ -47,3 +47,40 @@ __This is bold text.__
 		require.Equal(t, nothingChanges, stripped)
 	})
 }
+
+func TestHTML(t *testing.T) {
```
Comment (johnstcn) [inline coderd/parameter/renderer_test.go:51]: Can you add a test for script injection?
e.g. `<p onclick="alert(\"omghax\")">Click here to get access!</p>`?

File: cli/testdata/coder_server_--help.golden [hunk 2]
```diff
@@ -426,6 +426,10 @@ OIDC OPTIONS:
       --oidc-icon-url url, $CODER_OIDC_ICON_URL
           URL pointing to the icon to use on the OpenID Connect login button.
 
+      --oidc-signups-disabled-text string, $CODER_OIDC_SIGNUPS_DISABLED_TEXT
+          The custom text to show on the error page informing about disabled
+          OIDC signups.
```
Comment (johnstcn) [inline cli/testdata/coder_server_--help.golden:431]: This seems to support both markdown and HTML? Should we mention this or add some additional documentation on what is and is not allowed here?

**Review by johnstcn — COMMENTED** [review]
The only possible concern here is a bad actor injecting malicious HTML into the page. However in order to do this they have to get access to modify Coder's deployment values, and if this is the case then the deployment should already be considered compromised.

However, it's no harm to add some tests to validate that basic code injection is handled properly.

File: cli/testdata/coder_server_--help.golden (hunk 2 above)
Comment (mtojek) [reply cli/testdata/coder_server_--help.golden:431]: I disabled HTML tags on purpose, only Markdown content is rendered and every HTML is stripped. I can add a comment to `--help`.

**Review by johnstcn — APPROVED** [review]

File: cli/testdata/coder_server_--help.golden (hunk 2 above)
Comment (mtojek) [reply cli/testdata/coder_server_--help.golden]: Done

**Review by matifali — APPROVED** [review]

**Pull request merged**